tensorflow/contrib/image/python/ops/image_ops.py

```diff
@@ -508,6 +511,82 @@ def has_zero():
     else:
       return components
 
+def recursive_fill(mask, threshold, total_dims, max_iter=None):
+  """Recursively iterates until the provided 0-1s 2D or 3D mask is filled.
+  If max_iter is provided, it will iterate for a number of times.
+  If total_dims is 2, threshold is recommended to be 4.
+  If total_dims is 3, threshold is recommended to be 13.
+
+  Args:
+    mask: A 2D or 3D `Tensor` of 0s and 1s representing the mask to be filled.
```

Comment:
We should always allow 4-dimensional tensors, where the 0th axis contains a batch of images to be processed independently. The docstring should also explicitly state that a 3D or 4D input is always treated as containing voxels along the last 3 dimensions, which are always filled in 3-dimensional space.